src/token-cache.ts

```diff
@@ -147,27 +166,34 @@ class TokenCache {
   private validateToken(tokenName: string): Promise<Token> {
 
     if (this.tokenConfig[tokenName] === undefined) {
-      return Promise.reject(`Token ${tokenName} does not exist.`);
+      const msg = `TokenCache miss: ${tokenName} does not exist`;
+      this.logger.warn(msg);
+      return Promise.reject(msg);
     }
 
     const token = this._tokens[tokenName];
 
     if (token === undefined) {
-      return Promise.reject(`No token available for ${tokenName}`);
+      const msg = `TokenCache miss: ${tokenName}`;
+      this.logger.warn(msg);
+      return Promise.reject(msg);
     }
 
     const localExpiry = token.local_expiry || 0;
 
     if (localExpiry < Date.now()) {
-      return Promise.reject(`Token ${tokenName} expired locally.`);
+      const msg = `TokenCache miss: ${tokenName} expired locally`;
+      this.logger.warn(msg);
```

Comment:
really something we have to warn?

Reply:
Mh yeah you are right, maybe rather `debug` right?

Reply:
👍